cmd/frontend/auth/user.go

```diff
@@ -120,9 +123,18 @@ func GetAndSaveUser(ctx context.Context, db dbutil.DB, op GetAndSaveUserOp) (use
 			log15.Error("Failed to grant user pending permissions", "userID", userID, "error", err)
 		}
 
+		serviceTypeArg := json.RawMessage(fmt.Sprintf(`{"serviceType": %s}`, op.ExternalAccount.ServiceType))
```

Comment:
Hmm, seems like this won't generate a valid JSON, which will look like:

```json
`{"serviceType": github}`
```

I think what you want is:

```suggestion
		serviceTypeArg := json.RawMessage(fmt.Sprintf(`{"serviceType": %q}`, op.ExternalAccount.ServiceType))
```

Which quotes the string automatically.